src/components/NoteCard/NoteCard.jsx

```diff
@@ -1,24 +1,41 @@
 import React from 'react'
 
+
+import { BsFillTrashFill } from 'react-icons/bs';
+
+
 import './notecard.css'
 import { useNote } from '../../context/NoteContext'
+import { useModal } from '../../context/ModalContext'
 const NoteCard = ({noteobj}) => {
+    const {setmodalshow} = useModal();
     const {seteditnote,notes,setnotes} = useNote();
     const deleteNoteHandler = () =>{
         setnotes(notes.filter((obj)=>obj!==noteobj))
     }
     const editNoteHandler = () =>{
-        seteditnote({title:noteobj.title,data:noteobj.data,pinned:noteobj.pinned})
+        seteditnote({title:noteobj.title,data:noteobj.data,pinned:noteobj.pinned,archive:noteobj.archive,trash:noteobj.trash})
```

Comment:
all the property names are same as noteobj, so you can directly send the noteobj instead of creating a new object or use destructuring to create the new obj because the properties are same

Reply:
fixed this repetitive things by changing code note to object.